Automattic/woocommerce-payments
# Fix missing customer information after guest checkout via Checkout Block

Fixes #2296 

#### Changes proposed in this Pull Request

When checking out as a guest with checkout block customer information is not captured and payment methods are created on Stripe with empty billing address fields. This change fixes this by adding `billingData` as a dependency to the `onPaymentProcessing` effect so that it registers updates to the `billingData` fields.








#### Testing instructions





1. Install the [WooCommerce Blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/) plugin
2. Create a new page and add the Checkout block to that page
3. Open up a new private browser window (or any other form of a new session)
4. Navigate to the new block-based checkout page
5. Fill in the required information to checkout
6. Navigate to Payments > Transactions in wp-admin
7. Ensure that customer information associated with the transaction is visible in the transaction list and the transaction details.
8. Click the order number in the transactions list to navigate to the order associated with the transaction
9. Ensure that the customer information is present


![Screen Shot 2021-10-01 at 1 54 39 PM](https://user-images.githubusercontent.com/6216000/135672571-3225a054-a71f-40b9-90b5-6d27bae0b796.png)

-------------------


**Post merge**
